core/src/test/java/com/google/errorprone/bugpatterns/MemberNameTest.java

```diff
@@ -463,4 +463,32 @@ public void methodReference() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void methodNameWithMatchingReturnType() {
+    refactoringHelper
+        .addInputLines(
+            "Test.java",
+            "class Test {",
+            "  private Object Object() {",
+            "    return null;",
+            "  }",
+            "",
+            "  void call() {",
+            "     Object();",
+            "  }",
+            "}")
+        .addOutputLines(
+            "Test.java",
+            "class Test {",
+            "  private Object object() {",
```

Comment:
For other reviewers: before this fix Error Prone would suggest:
```suggestion
            "  private object Object() {",
```